### PR TITLE
Support tensorflow 2.2

### PR DESCRIPTION
**Pull request by jiafatom**
This is from [PR](https://github.com/onnx/keras-onnx/pull/482) with changes addressing reviewer's comments.

### CODE REVIEW

File: keras2onnx/parser.py [hunk 1]
```diff
@@ -482,20 +482,28 @@ def _get_output_nodes(node_list, node):
     return [n_ for n_ in node_list if n_ not in nodes_has_children]  # need to keep the order.
 
 
+def _filter_out_input(node_name):
+    # tf.keras BN layer sometimes create a placeholder node 'scale' in tf2.x. 
+    # Given bn layer will be converted in a whole layer, it's fine to just filter this node out.
+    filter_out = re.match(r"batch_normalization_\d+\/scale$", node_name)
```
Comment (wenbingl) [inline keras2onnx/parser.py:488]: Put all patterns in a list to be easily extended.

File: keras2onnx/parser.py [hunk 2]
```diff
@@ -482,20 +482,28 @@ def _get_output_nodes(node_list, node):
...
+            if not _filter_out_input(predecessor.name):
                 inputs.add(predecessor)
                 graph_inputs.add(predecessor)
+                continue
         if predecessor in layer_nodes or len(layer_nodes) == 0:
             subgraph.append(predecessor)
         else:
-            inputs.add(predecessor)
-            q_overall.put_nowait(predecessor)
+            if not _filter_out_input(predecessor.name):
```
Comment (wenbingl) [inline keras2onnx/parser.py:504]: Can you confirm why some many filters needed?

File: keras2onnx/parser.py [hunk 3]
```diff
@@ -735,7 +743,15 @@ def parse_graph_modeless(topo, graph, target_opset, input_names, output_names, k
 
     for ts_i_ in input_tensors:
         var_type = _adjust_input_batch_size(infer_variable_type(ts_i_, target_opset))
-        str_value = ts_i_.name
+        if ts_i_.name.endswith(':0'):
```
Comment (wenbingl) [inline keras2onnx/parser.py:746]: Any special things in the model causes ':0'?

File: keras2onnx/parser.py (hunk 2 above)
Comment (jiafatom) [reply keras2onnx/parser.py]: This seems redundant, removed.

File: keras2onnx/parser.py (hunk 3 above)
Comment (jiafatom) [reply keras2onnx/parser.py:747]: This is tensor name so it has ":0". This code path is `parse_graph_modeless`, in the traditional path, we also have logic to handle this, see [here](https://github.com/onnx/keras-onnx/blob/89b0c459ea2dd1aa086d16983d70547992ef8095/keras2onnx/parser.py#L775-L785)

**Pull request merged**